copydb/config.go

```diff
@@ -107,6 +107,11 @@ func (c *Config) InitializeAndValidateConfig() error {
 		}
 	}
 
+	// Don't enforce that there are no foreign keys present if TablesToBeCreatedFirst is specified.
+	if len(c.TablesToBeCreatedFirst) > 0 {
+		c.SkipForeignKeyConstraintsCheck = true
+	}
```

Comment:
Not sure if we should have this implicit behavior.

Reply:
I agree, this conditional was definitely a code smell. 

Right now, `SkipForeignKeyConstraintsCheck` lives on the ferry, while `TablesToBeCreatedFirst` lives at the `copydb` level of abstraction. There's an inherent design boundary here.

It appears there are two options given this setup:

1. Add the implicit behaviour at the `copydb` level to set `SkipForeignKeyConstraintsCheck` (current)
2. Let the end user of the `copydb` manually configure `SkipForeignKeyConstraintsCheck` when they use `TablesToBeCreatedFirst`. Perhaps I can add a comment in the description of `TablesToBeCreatedFirst`.

There's downsides to both. In one case, there's implicit behaviour. In the other, we create a sense of coupling between two unrelated config options.

I'll go with option 2, because there _may_ be a use case of `TablesToBeCreatedFirst`, that doesn't necessarily relate to foreign key constraints.

Thoughts?

Reply:
4434cc8

Reply:
I would go with option 2.